packages/astro/src/@types/astro.ts

```diff
@@ -72,21 +72,101 @@ export interface BuildConfig {
 }
 
 /**
- * Astro.* available in all components
- * Docs: https://docs.astro.build/reference/api-reference/#astro-global
+ * Astro global available in all contexts in .astro files
+ *
+ * [Astro documentation](https://docs.astro.build/reference/api-reference/#astro-global)
  */
 export interface AstroGlobal extends AstroGlobalPartial {
-	/** get the current canonical URL */
+	/** Canonical URL of the current page. If the [site](https://docs.astro.build/en/reference/configuration-reference/#site) config option is set, its origin will be the origin of this URL.
+	 *
+	 * [Astro documentation](https://docs.astro.build/en/reference/api-reference/#astrocanonicalurl)
+	 */
 	canonicalURL: URL;
-	/** get page params (dynamic pages only) */
+	/** Parameters passed to a dynamic page generated using [getStaticPaths](https://docs.astro.build/en/reference/api-reference/#getstaticpaths)
+	 *
+	 * Example usage:
+	 * ```astro
+	 * ---
+	 * export async function getStaticPaths() {
+	 *    return [
+	 *     { params: { id: '1' } },
+	 *   ];
+	 * }
+	 *
+	 * const { id } = Astro.params;
+	 * ---
+	 * <h1>{id}</h1>
+	 * ```
+	 *
+	 * [Astro documentation](https://docs.astro.build/en/reference/api-reference/#params)
+	 */
 	params: Params;
-	/** set props for this astro component (along with default values) */
+	/** List of props passed to this component
+	 *
+	 * A common way to get specific props is through [destructuring](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment), ex:
+	 * ```javascript
+	 * const { name } = Astro.props
+	 * ```
+	 *
+	 * [Astro documentation](https://docs.astro.build/en/core-concepts/astro-components/#component-props)
+	 */
 	props: Record<string, number | string | any>;
-	/** get information about this page */
+	/** Information about the current request. This is a standard [Request](https://developer.mozilla.org/en-US/docs/Web/API/Request) object
+	 *
+	 * For example, to get an URL object of the current URL, you can use:
+	 * ```javascript
+	 * const url = new URL(Astro.request.url);
+	 * ```
+	 *
+	 * [Astro documentation](https://docs.astro.build/en/reference/api-reference/#astrorequest)
+	 */
 	request: Request;
-	/** see if slots are used */
+	/** Redirect to another page (**SSR Only**)
+	 *
+	 * Example usage:
+	 * ```typescript
+	 * if(!isLoggedIn) {
+	 *   return Astro.redirect('/login');
+	 * }
+	 * ```
+	 *
+	 * [Astro documentation](https://docs.astro.build/en/guides/server-side-rendering/#astroredirect)
+	 */
+	redirect(path: string): Response;
+	/** Utility functions for modifying an Astro component’s slotted children
```

Comment:
Since writing just `Astro.slots` is rare, I just copied the documentation and didn't elaborate further outside of a link

I know this technically does not document what `Astro.slots` really is, but our documentation also doesn't, so I figured we might not want users to use it directly (instead preferring the helper methods)